Firestore/Source/API/FIRQuery.mm

```diff
@@ -348,7 +349,13 @@ - (FIRQuery *)queryOrderedByFieldPath:(FIRFieldPath *)fieldPath descending:(BOOL
 }
 
 - (FIRQuery *)queryLimitedTo:(NSInteger)limit {
-  return Wrap(_query.Limit(static_cast<int64_t>(limit)));
+  int32_t internalLimit;
+  if (limit == NSNotFound || limit > std::numeric_limits<int32_t>::max()) {
```

Comment:
Hrm. On JS / Android we reject non-positive limits, e.g. https://github.com/firebase/firebase-js-sdk/blob/31438c18e7c525bf6181fa75826e8b87d7b61bb6/packages/firestore/src/api/database.ts#L1538

We should ideally do the same here rather than treat -1 as "no limit"...  But I guess this is the (accidental) preexisting behavior.  If we're worried about people relying on this undocumented behavior, we should at least have a porting note explaining why iOS is different.

Reply:
Ha. This change passed unit tests but not integration tests.

Prior to this change this was actually enforced, just not here:

```
caught "FIRInvalidArgumentException", "Invalid Query. Query limit (-1) is invalid. Limit must be positive."
(
	0   CoreFoundation                      0x000000010b52b1bb __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x000000010aac9735 objc_exception_throw + 48
	2   Firestore_Example_iOS               0x00000001080ea024 _ZN8firebase9firestore3api4impl20ThrowInvalidArgumentERKNSt3__112basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEE + 52
	3   Firestore_Example_iOS               0x0000000107f5b17e _ZN8firebase9firestore3api20ThrowInvalidArgumentIJxEEEvPKcDpRKT_ + 46
	4   Firestore_Example_iOS               0x0000000108166bde _ZNK8firebase9firestore3api5Query5LimitEx + 62
	5   Firestore_Example_iOS               0x0000000107f607ae -[FIRQuery queryLimitedTo:] + 62
	6   Firestore_IntegrationTests_iOS      0x000000012b174545 -[FIRQueryTests testNegativeLimitQueries] + 725
```

Using `-1` for `Query::kNoLimit` doesn't work :-).

PTAL for an alternate implementation.